flfm/app/main.py

```diff
@@ -0,0 +1,470 @@
+        data = flfm_io.open(bytestream)
+        logger.info(f"Uploading {filename} completed.")
+        return data
+
+
+@callback(
+    Output(dict(type="display", index="reconstruction"), "children", allow_duplicate=True),
+    Input("run-reconstruction", "n_clicks"),
+    Input("rl-iters", "value"),
+    State("center_x", "value"),
+    State("center_y", "value"),
+    State("radius", "value"),
+    prevent_initial_call=True,
+)
+def reconstruct(n_clicks, rl_iters, center_x, centre_y, radius):
+    """RUN FLFM reconstruction and return reconstructed image."""
+    global image_data
+    if n_clicks == 0:
+        return no_update
+
+    if image_data["psf"] is None or image_data["image"] is None:
+        raise no_update
+
+    id = "reconstruction"
+    image_data["uncropped_reconstruction"] = flfm_restoration.richardson_lucy(
+        image_data["image"], image_data["psf"], num_iter=rl_iters
+    )
+    image_data[id] = flfm.util.crop_and_apply_circle_mask(
+        image_data["uncropped_reconstruction"],
+        center=(center_x, centre_y),
+        radius=radius,
+    )
+    children, _n_frames = _display_image_with_slider(image_data[id], id=id)
+    return children
+
+
+@callback(
+    Output(dict(type="image-graph", index="psf"), "figure", allow_duplicate=True),
+    Output("normalize-psf", "disabled"),
+    Input("normalize-psf", "n_clicks"),
+    State(dict(type="image-slider", index="psf"), "value"),
+    State(dict(type="color-scale", index="psf"), "value"),
+    prevent_initial_call=True,
+)
+def normalize_psf(n_clicks, frame, color_scale):
+    """Normalize PSF and disable button once normalized."""
+    global image_data
+    if n_clicks == 0 or image_data["psf"] is None:
+        return no_update
+
+    id = "psf"
+    image_data[id] = image_data[id] / psf_sum(image_data[id])
+    fig = plot_image(image_data[id][frame, :, :], color_scale=color_scale)
+    return fig, True
+
+
+@callback(
+    Output(dict(type="image-graph", index="reconstruction"), "figure", allow_duplicate=True),
+    Input("center_x", "value"),
+    Input("center_y", "value"),
+    Input("radius", "value"),
+    State(dict(type="image-slider", index="reconstruction"), "value"),
+    prevent_initial_call=True,
+)
+def update_crop(center_x, center_y, radius, frame):
+    if image_data["uncropped_reconstruction"] is None:
+        return no_update
+
+    image_data["reconstruction"] = flfm.util.crop_and_apply_circle_mask(
+        image_data["uncropped_reconstruction"], center=(center_x, center_y), radius=radius
+    )
+    fig = plot_image(image_data["reconstruction"][frame, :, :])
+    return fig
+
+
+@callback(
+    Output(dict(type="image-graph", index=MATCH), "figure", allow_duplicate=True),
+    Input(dict(type="color-scale", index=MATCH), "value"),
+    State(dict(type="image-slider", index=MATCH), "value"),
+    State(dict(type="color-scale", index=MATCH), "id"),
+    prevent_initial_call=True,
+)
+def update_color_scale(color_scale, frame, id):
```

Comment:
Can you add some type hints here if the types are consistent and reasonable to reference?